### DISCUSSION TITLE
Future of eslintrc validations

### DISCUSSION THREAD

**Discussion by ayroblu** [eslint/eslint#15605] [Q&A]
Hi! I was wondering about best practice and future of eslintrc validations?

As from: #12078, it looks like eslint maintainers are actively against adding TypeScript support for `.eslintrc.ts`.

There is also json schema files you can use, such as in https://github.com/SchemaStore/schemastore/blob/master/src/schemas/json/eslintrc.json and these pretty much just work if you have it configured, but from my testing, they don't validate typos, but do provide suggestions for basic syntax.

If you're willing to setup your own pipeline, then you can do this:
```typescript
import type { TSESLint } from "@typescript-eslint/experimental-utils";

const config: TSESLint.Linter.Config = {
  ...
}
export default config;
```
Which is great but the types aren't that great, e.g. making a typo in a rule name won't be caught

This is the great paradox of eslint, is there no good linting for the eslintrc?

Ideal scenario:
1. eslintrc file linted, meaning if I make a typo in one of the params, or a plugin name, or parserOptions etc, it would be caught
2. Rules are typed, meaning it's obvious what all the different extra params are
2. Documentation in the types to explain what they do (this can be seen in some of the json schema ones)

**Comment by nzakas** [comment]
eslintrc will be going away in the near future in favor of a new config system, so we aren’t making any further changes to eslintrc at this time.
